analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -17,101 +17,59 @@
 
 from codechecker_common.logger import get_logger
 
+from codechecker_merge_ctu_func_maps.merge_ctu_func_maps import merge
+
 from .. import analyzer_base
 from . import ctu_triple_arch
 
 LOG = get_logger('analyzer')
 
 
-def generate_func_map_lines(fnmap_dir):
-    """ Iterate over all lines of input files in random order. """
-
-    files = glob.glob(os.path.join(fnmap_dir, '*'))
-    for filename in files:
-        with open(filename, 'r', encoding='utf-8',
-                  errors="ignore") as in_file:
-            for line in in_file:
-                yield line
-
-
-def create_global_ctu_function_map(func_map_lines):
-    """ Takes iterator of individual function maps and creates a global map
-    keeping only unique names. We leave conflicting names out of CTU.
-    A function map contains the id of a function (mangled name) and the
-    originating source (the corresponding AST file) name."""
-
-    mangled_to_asts = {}
-
-    for line in func_map_lines:
-        mangled_name, ast_file = line.strip().split(' ', 1)
-        # We collect all occurences of a function name into a list
-        if mangled_name not in mangled_to_asts:
-            mangled_to_asts[mangled_name] = {ast_file}
-        else:
-            mangled_to_asts[mangled_name].add(ast_file)
-
-    mangled_ast_pairs = []
-
-    for mangled_name, ast_files in mangled_to_asts.items():
-        if len(ast_files) == 1:
-            mangled_ast_pairs.append((mangled_name, ast_files.pop()))
-
-    return mangled_ast_pairs
+def merge_ctu_func_maps(ctu_dir, ctu_func_map_file, ctu_temp_fnmap_folder):
+    """ Merge individual function maps into a global one."""
 
+    triple_arches = glob.glob(os.path.join(ctu_dir, '*'))
```

Comment:
Shouldn't this logic be in the merge_ctu_func_maps modul, so the merging is done based on the arch triple?
Does the merge tool know about the triple arch automatically based on the output, or that information is only visible in the used directory name?

Reply:
For me this function is a little bit CodeChecker specific, and the merge_ctu_func_maps tool work only with a function map directory and not with a ctu dir. Later we can extend the tool if we want.

Reply:
I understand that but the merge_ctu_func maps tool does not work on the ctu directories, it merges the extdef outputs into one which can be used by the ctu during analysis.
I still think this merging should be possible without codechecker, if I generate the extdef files without codechecker and I want to merge them into one this tool should give that feature. The exdef tool is independent from the ctu in that sense that only the output of that separate binary is used by the ctu analysis.